src/components/common/NameRequestInfo.vue

```diff
@@ -61,34 +62,115 @@
+          id='name-translation-checkbox'
+          label="This company uses one of more translations of its name outside of Canada."
+          @click.native="confirmNameTranslation()"
+        />
+        <template v-if="hasNameTranslation">
+          <p><b>Note:</b> Name translations must use the Latin
+            Alphabet (English, French, etc.). Names that use other writing systems must spell the name phonetically
+            in English or French.
+          </p>
+          <v-btn outlined color="primary" @click="isAddingNameTranslation = true" :disabled="isAddingNameTranslation">
+            <v-icon>mdi-plus</v-icon>
+            <span>Add a Name Translation</span>
+          </v-btn>
+        </template>
+      </v-col>
+    </v-row>
+    <!-- Name Translation Components -->
+    <v-row v-if="hasNameTranslation" id="name-translation-container">
+      <!-- Spacer Column -->
+      <v-col></v-col>
+      <v-col>
+        <add-name-translation
+          v-if="isAddingNameTranslation"
+          :edit-name-translation="editingNameTranslation"
+          @addTranslation="addName($event)"
+          @cancelTranslation="cancelNameTranslation()"
+          @removeTranslation="removeNameTranslation(editIndex)"
+        />
+        <list-name-translations
+          v-if="getNameTranslations && getNameTranslations.length"
+          :isAddingNameTranslation="isAddingNameTranslation"
+          :translationsList="getNameTranslations"
+          @editNameTranslation="editNameTranslation($event)"
+          @removeNameTranslation="removeNameTranslation($event)"
+        />
+      </v-col>
+    </v-row>
   </div>
 </template>
 
 <script lang="ts">
 // Libraries
-import { Component, Mixins, Vue } from 'vue-property-decorator'
-import { Getter } from 'vuex-class'
+import { Component, Emit, Mixins } from 'vue-property-decorator'
+import { Action, Getter, State } from 'vuex-class'
 import { getName } from 'country-list'
 
+// Components
+import { ConfirmDialog } from '@/components/dialogs'
+import { ListNameTranslations, AddNameTranslation } from '@/components/DefineCompany'
+
 // Enums
 import { NameRequestStates } from '@/enums'
 
 // Interfaces
-import { GetterIF, NameRequestDetailsIF, NameRequestApplicantIF } from '@/interfaces'
+import {
+  GetterIF,
+  NameRequestDetailsIF,
+  NameRequestApplicantIF,
+  ConfirmDialogType,
+  ActionBindingIF
+} from '@/interfaces'
 
 // Mixins
 import { DateMixin } from '@/mixins'
 
-@Component({})
+@Component({
+  components: {
+    ListNameTranslations,
+    AddNameTranslation,
+    ConfirmDialog
+  }
+})
 export default class NameRequestInfo extends Mixins(DateMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationRemovalDialog: ConfirmDialogType
+  }
+
   // Template Enums
   readonly NameRequestStates = NameRequestStates
 
+  // Local Properties
+  private hasNameTranslation = false
+  private isAddingNameTranslation = true
+  private editingNameTranslation = ''
+  private editIndex: number
```

Comment:
Should this be initialized? As-is it's undefined, but maybe -1 is a better initial value? Just wondering.

Reply:
That makes sense for sure. I set it -1 before its used anyways, might as well initialize it!